src/oneD/Flow1D.cpp

```diff
@@ -827,37 +842,48 @@ string Flow1D::componentName(size_t n) const
     }
 }
 
-size_t Flow1D::componentIndex(const string& name) const
+size_t Flow1D::componentIndex(const string& name, bool checkAlias) const
 {
-    if (name=="velocity") {
-        return c_offset_U;
-    } else if (name=="spread_rate") {
-        return c_offset_V;
-    } else if (name=="T") {
-        return c_offset_T;
-    } else if (name=="lambda") {
-        return c_offset_L;
-    } else if (name == "eField") {
-        return c_offset_E;
-    } else if (name == "Uo") {
-        return c_offset_Uo;
-    } else {
-        for (size_t n=c_offset_Y; n<m_nsp+c_offset_Y; n++) {
-            if (componentName(n)==name) {
-                return n;
-            }
+    if (componentMap.count(name)) {
+        return componentMap.at(name);
+    }
+    for (size_t n=c_offset_Y; n<m_nsp+c_offset_Y; n++) {
+        if (componentName(n)==name) {
+            return n;
+        }
+    }
+    if (checkAlias) {
+        auto aliasMap = _componentAliasMap();
+        if (aliasMap.count(name)) {
+            return componentIndex(aliasMap.at(name), false);
+        }
+    }
+    throw CanteraError("Flow1D::componentIndex",
+                       "No component named '{}'", name);
+}
+
+bool Flow1D::hasComponent(const string& name, bool checkAlias) const
+{
+    if (componentMap.count(name)) {
+        return true;
+    }
+    for (size_t n=c_offset_Y; n<m_nsp+c_offset_Y; n++) {
+        if (componentName(n)==name) {
```

Comment:
Very minor, but I think I'd prefer
```suggestion
    for (size_t n = c_offset_Y; n < m_nsp+c_offset_Y; n++) {
        if (componentName(n) == name) {
```
here and elsewhere. We're reasonably consistent about having space around the binary comparison operators and the assignment operator in `for` loops.